pkg/kubectl/cmd/apply.go

```diff
@@ -61,6 +61,8 @@ const (
 	backOffPeriod = 1 * time.Second
 	// how many times we can retry before back off
 	triesBeforeBackOff = 1
+
+	warningNoLastAppliedConfigAnnotation = "Warning: kubectl apply should be used on resource created by eithor kubectl create --save-config or kubectl apply\n"
```

Comment:
The moment the user see this warning, it's already too late. Therefore, we should tell them what to expect in this case, or show them how to react.  

We should probably generate an error instead, to stop the users from doing this. Is this considered a breaking change? @kubernetes/kubectl